src/Catalog.API/Model/CatalogServices.cs

```diff
@@ -1,11 +1,12 @@
 ﻿using eShop.Catalog.API.Services;
+using Microsoft.AspNetCore.Mvc;
 
 public class CatalogServices(
     CatalogContext context,
-    ICatalogAI catalogAI,
+    [FromServices] ICatalogAI catalogAI,
```

Comment:
Why is an explicit `[FromServices]` needed here?

Reply:
Because at build time we bypass registering the services and then model-binding doesn't know what do to with these parameters. An alternative was to register dummy instances of these services at build time but this seemed like the cleaner approach.